src/components/common/DiscountCode.vue

```diff
@@ -0,0 +1,137 @@
+<template>
```

Comment:
Any component that contains a form, we have been calling them with the suffix `Form` to quickly identify them (and has proven to be quite useful for me). Also, this component is only for adding discount code, as far as I see. So this one would be better called `AddDiscountCodeForm.vue`, in my opinion.